### DISCUSSION TITLE
Unable to browse OPC DA server

### DISCUSSION THREAD

**Discussion by djerridjerri** [duduyoyo/WebSocket4OPC#10] [Q&A] [answered]
Hi,

I am writing here hoping to get some bright ideas or solutions to the problem I am facing.  I think I have correctly installed WebSocket4OPC - I can create a simulation server using MatrikonOPC simulation server and access that server using websocket.html - here I sent the browse command, getting the expected results:
![browse simulation server 1](https://github.com/duduyoyo/WebSocket4OPC/assets/116136245/2cf765c8-7dba-4c62-8916-daeaf9562c2b)

I have 3 different OPC servers running on this box, I can see them listed in Matrikon's explorer software.  I can also read all the servers using the explorer software.

![matrikon explorer 1](https://github.com/duduyoyo/WebSocket4OPC/assets/116136245/19a20cc6-7661-4654-adc0-991327f99a01)

I'd like to control the ThermoElectron ones, so I change the config file from the simulation server to:

```
<Config>
	<DAProgID>ThermoElectron.RESULTOPCServerDA.2</DAProgID>
	<AEProgID>ThermoElectron.RESULTOPCServerEA.1</AEProgID>
</Config>
```

And then I head over to websocket.html and sending browse gives me:

![browse thermo server 1](https://github.com/duduyoyo/WebSocket4OPC/assets/116136245/79697cf9-5d09-4278-b8c1-1ee035275d06)

And I don't know what's wrong.  I don't know why Matrikon's explorer can read the tags in the ThermoElectron servers and WebSocket4OPC can't.

Any advice would be appreciated.
Thank you.

**Comment by duduyoyo** [comment]
I need your screenshots from Event Viewer, or you can send compiled log info from Event Viewer to websocket4opc@gmail.com. At least I can see there is a typo for AE prog ID from above. Cheers!

**Reply by djerridjerri** [reply]
Repeated the tests above after fixing the typo, unfortunately that doesn't seem to have solved the problem.
Sent you an email with the Event Viewer information.  Cheers!

**Reply by duduyoyo** [reply, marked as answer]
That is because Thermo Electron DA server complies with DA spec 2.0 and Websocket4OPC complies with spec 3.0. Now I have ported back to support 2.0. You can get the latest download to verify. Thanks!

**Comment by duduyoyo** [comment]
Sent you notes via email (on 4/24) on testing in your box. Please let me know after your test. Thanks!
